internal/backends/collection.go

```diff
@@ -105,17 +105,29 @@ type QueryResult struct {
 // It also can be used to close the returned iterator and free underlying resources,
 // but doing so is not necessary - the handler will do that anyway.
 //
-// If SortPushdown is false, any sort, or limit pushdown won't be performed.
-// If SortPushdown is false, Sort and Limit fields cannot be set.
-// If SortPushdown is true, but Sort is equal to nil, sort pushdown can still occure
+// - If SortPushdown is false, any sort, or limit pushdown won't be performed.
+// - If SortPushdown is false, Sort and Limit fields cannot be set.
+// - If SortPushdown is true, but Sort is equal to nil, sort pushdown can still occure
 // for capped collection if recordID is not set.
+//
+// - If Filter is provided, the pushdown will be performed if supported.
+// Results may contain more documents than necessary, those should be filtered out at the handler level.
+// - If Sort is provided, the unsafe sort pushdown will be performed;
+// - If Limit is provided standalone, the safe sort pushdown will be performed;
+// - If Filter and Sort are provided, the unsafe sort pushdown for both will be performed if supported;
```

Comment:
And if not? How will the handler know if sorting was performed and if there is no need to fetch all documents and sort them?

Reply:
I think me might want to introduce such information in `QueryResult`

Reply:
So, why did you re-request the review?